linker error when using module as a test dependency

I get linker errors on x86-linux-native when using cpputest as a test framework. If I make cpputest as a direct dependency, the tests get built properly, but not if I turn them into a targetDepency.

Here's the sort of linker error I get:

```
info: generate for target: x86-linux-native 0.0.6 at /tmp/cpputest/yotta_targets/x86-linux-native
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/cpputest/build/x86-linux-native
[5/5] Linking CXX executable test/mbed-client-trace-test-firsttest
FAILED: : && /usr/bin/c++   -O2 -g -DNDEBUG   test/CMakeFiles/mbed-client-trace-test-firsttest.dir/tmp/cpputest/test/FirstTest/FirstTest.cpp.o test/CMakeFiles/mbed-client-trace-test-firsttest.dir/tmp/cpputest/test/FirstTest/main.cpp.o  -o test/mbed-client-trace-test-firsttest  -rdynamic ym/cpputest/source/libcpputest.a yotta_dummy_lib_mbed_client_trace/libmbed-client-trace.a && :
test/CMakeFiles/mbed-client-trace-test-firsttest.dir/tmp/cpputest/test/FirstTest/FirstTest.cpp.o:(.rodata._ZTV39TEST_FirstTestGroup_FirstTest_TestShell[_ZTV39TEST_FirstTestGroup_FirstTest_TestShell]+0x30): undefined reference to `UtestShell::isNull() const'
```

Here's the module.json which uses cpputest as a test-dependency.
```
{
  "name": "mbed-client-trace",
  "version": "0.0.1",
  "description": "Simple Trace library for mbed clients",
  "keywords": [
    "trace",
    "debug",
    "library",
    "client"
  ],
  "author": "Jussi Vatjus-Anttila",
  "repository": {
    "url": "https://github.com/ARMmbed/mbed-client-trace.git",
    "type": "git"
  },
  "homepage": "https://github.com/ARMmbed/mbed-client-trace",
  "licenses": [
    {
      "url": "https://spdx.org/licenses/Apache-2.0",
      "type": "Apache-2.0"
    }
  ],
  "dependencies": {
  },
  "testDependencies": {
    "cpputest": "*"
  }
}
```

The cpputest module on github/armmbed doesn't support the gcc platform. I'll send you a tar ball for what I've got.

--- a/yotta/lib/cmakegen.py
+++ b/yotta/lib/cmakegen.py
@@ -37,6 +37,7 @@ def sanitizeSymbol(sym):
 jinja_environment.filters['replaceBackslashes'] = replaceBackslashes
 jinja_environment.filters['sanitizePreprocessorSymbol'] = sanitizePreprocessorSymbol
 jinja_environment.globals['list'] = list
+jinja_environment.globals['pathJoin'] = os.path.join
 
 class SourceFile(object):
     def __init__(self, fullpath, relpath, lang):

--- a/yotta/lib/templates/test_CMakeLists.txt
+++ b/yotta/lib/templates/test_CMakeLists.txt
@@ -10,13 +10,13 @@ include_directories("{{ source_directory | replaceBackslashes }}")
 
 {% for component in test_dependencies %}
     {% for d in component.getExtraSysIncludes() %}
-        include_directories(SYSTEM "{{ d | replaceBackslashes }}")
+        include_directories(SYSTEM "{{ pathJoin(component.path, d) | replaceBackslashes }}")
     {% endfor %}
 {% endfor %}
 
 {% for component in test_dependencies %}
     {% for d in component.getExtraIncludes() %}
-        include_directories("{{ d | replaceBackslashes }}")
+        include_directories("{{ pathJoin(component.path, d) | replaceBackslashes }}")
     {% endfor %}
 {% endfor %}
 

